assets/javascripts/discourse/components/chat-live-pane.js

```diff
@@ -410,6 +418,23 @@ export default Component.extend({
     return message.id || `staged-${message.stagedId}`;
   },
 
+  _getLastReadId() {
+    return this.currentUser.chat_channel_tracking_state[this.chatChannel.id]
+      ?.chat_message_id;
+  },
+
+  _includeInmediateHistory(findArgs) {
+    let lastReadId = this._getLastReadId();
+
+    if (lastReadId) {
+      // We are fetching the last read message, the two previous ones,
+      // and the next 47 to complete a page.
+      const offset = 47;
```

Comment:
Maybe we should use `50 - previousMessagesToDisplay` here. Actually, `50` should be moved to a page size constant as well since it is a param.